### DISCUSSION TITLE
.NET 8.0 and VS 17.8 Stable support

### DISCUSSION THREAD

**Discussion by jeromelaban** [unoplatform/uno#14427] [General]
.NET 8.0.100 and VS 17.8 has now been released, and we're working to get everything up and running for the Uno Platform community.

## Visual Studio support
At present time, there are no known issues when installing and using VS 17.8.

## .NET SDK
[Upgrading uno.check](https://platform.uno/docs/articles/external/uno.check/doc/using-uno-check.html) to 1.17.0 allows for .NET 8.0.100 and its workloads to be installed properly, on all platforms.

~~The .NET 8.0.100 SDK is having a layout issue with https://github.com/dotnet/sdk/issues/36884 which contains RC2 workload manifests and fails to install properly with uno.check (https://github.com/unoplatform/uno.check/pull/197). We're currently tracking the updates that will be posted by the .NET team so we can update everything and get everyone running.~~

~~Until this issue is resolved, installing VS 17.8 installs the 8.0.100 mobile workloads properly and if you're not running uno-check on Windows, you will be good to code. On macOS, until this is resolved, we do not have a workaround. Linux does not use workloads so installing .NET 8.0.100 manually will be enough to use VS Code and build wasm/gtk based apps.~~

**Comment by jeromelaban** [comment]
A new version of uno.check is now available for .NET 8.0.100, happy coding!

**Reply by TopperDEL** [reply]
My Uno-check reports green, I have VS 17.9 Preview running and just created a fresh Uno-App using the extension - after hitting F5 I get many messages like "The provided RuntimeIdentifier "win10-x86" is not recognized". I also get this on a working solution that I manually migrated to .Net 8. Any ideas? Is it just me and my installation?

![image](https://github.com/unoplatform/uno/assets/1833242/1d7142ec-5144-4503-b39b-84298a5e4d15)

**Reply by TopperDEL** [reply]
I just uninstalled and cleaned Visual Studio and .Net - after reinstalling everything and getting a green uno-check I still get the errors above.

A .Net-MAUI-app using .Net 8 is working.

**Reply by limeniye** [reply]
@TopperDEL, it looks like you are using the old uno solution template for .net 8.0.
As I know, it was already fixed -> https://github.com/unoplatform/uno.templates/issues/183
To fix it yourself, do the following replace:
![image](https://github.com/unoplatform/uno/assets/69314237/98310f8a-19fd-4fd3-9cb2-97ee91abfebd)

**Reply by TopperDEL** [reply]
That did not help, unfortunately. But it seems to be this machine - another machine works without problems. What I'm wondering: it references RuntimeIdentifiers not used in this project (and which cannot be found by searching).

If I clone my problematic solution to the other machine, I get the same errors. Weird. So my solution seems to have a problem. I will compare the project files with a working Net8-Uno.project, now.

**Reply by limeniye** [reply]
@TopperDEL, hm, do you have the PublishProfiles at your Windows project?
If not, create an empty Uno application and move this folder to your application. Inside of this files you can see the **RuntimeIdentifier**
![image](https://github.com/unoplatform/uno/assets/69314237/7425fbc4-a886-45f8-b781-7967c82ecd35)

>What I'm wondering: it references RuntimeIdentifiers not used in this project

I see that Windows project has a **RuntimeIdentifiers** 
![image](https://github.com/unoplatform/uno/assets/69314237/04e7c906-8765-4719-ae7a-97e4350d0d26)

Try to add this lines by yourself to Windows project file:
```
<Platforms>x86;x64;arm64</Platforms>
<RuntimeIdentifiers>win-x86;win-x64;win-arm64</RuntimeIdentifiers>
<PublishProfile>win-$(Platform).pubxml</PublishProfile>
```

**Reply by TopperDEL** [reply]
Thank you @limeniye! Adding the three lines to my "shared" project (the UnoApp34 in your screen above) did the trick. I'm not quite sure why I need it there. But I use a quite complex logic with Skia and other packages. But now my app is running with .Net 8! Thank you!

**Reply by limeniye** [reply]
@TopperDEL, hm. I'am very glad to help you. But I guess this is not quite correct.
Please, tell me what do you have at the end of **Directory.Build.props** file? 
Do you have this condition?
![image](https://github.com/unoplatform/uno/assets/69314237/8cb1d575-0c7a-42a5-9e66-282b8e0cbaf0)
```XML
    <When Condition="$(TargetFramework.Contains('windows10'))">
      <PropertyGroup>
        <IsWinAppSdk>true</IsWinAppSdk>
        <SupportedOSPlatformVersion>10.0.18362.0</SupportedOSPlatformVersion>
        <TargetPlatformMinVersion>10.0.18362.0</TargetPlatformMinVersion>
        <RuntimeIdentifiers>win-x86;win-x64;win-arm64</RuntimeIdentifiers>
        <EnableCoreMrtTooling Condition=" '$(BuildingInsideVisualStudio)' != 'true' ">false</EnableCoreMrtTooling>
      </PropertyGroup>
    </When>
```

**Reply by TopperDEL** [reply]
I have this one:

```
<When Condition="$(TargetFramework.Contains('windows10'))">
      <PropertyGroup>
        <IsWinAppSdk>true</IsWinAppSdk>
        <SupportedOSPlatformVersion>10.0.19041.0</SupportedOSPlatformVersion>
        <TargetPlatformMinVersion>10.0.19041.0</TargetPlatformMinVersion>
      </PropertyGroup>
    </When>
```
I will compare it with a working version from a fresh Uno-Net8-app.

**Reply by limeniye** [reply]
@TopperDEL, understood.
>Adding the three lines to my "shared" project (the UnoApp34 in your screen above) did the trick
- Remove this lines from the shared uno lib (UnoApp34) and add this three lines to the Windows project:
![image](https://github.com/unoplatform/uno/assets/69314237/89ca1498-d1c4-473a-9462-b942f38bff0e)
- Add the ```<RuntimeIdentifiers>win-x86;win-x64;win-arm64</RuntimeIdentifiers>``` to the your **"$(TargetFramework.Contains('windows10'))"** condition.  This should work.

Otherwise, if you added this to your shared uno lib without specifying that these lines should only be used for the WinUI project, this can cause problems at other projects.

**Reply by TopperDEL** [reply]
It's working, now. Thanks!
